lumicks/pylake/tests/test_imaging_camera/test_correlated_stack.py

```diff
@@ -927,3 +927,18 @@ def test_alignment_multistack(rgb_alignment_image_data, gray_alignment_image_dat
             full_stack_idx += 1
 
     assert full_stack.num_frames == len(ref_ts)
+
+
+def test_frame_timestamp_ranges_exclude_false():
+    stack = CorrelatedStack.from_dataset(
+        TiffStack(
+            [MockTiffFile(data=[np.ones((5, 5))] * 4, times=make_frame_times(4, step=6))],
+            align_requested=False,
+        )
+    )
+
+    for exclude, ranges in zip(
+            [False, True],
+            [[(10, 20), (20, 30), (30, 40), (40, 46)], [(10, 16), (20, 26), (30, 36), (40, 46)]],
```

Comment:
If you calculate the last timestamp range

```suggestion
            [[(10, 20), (20, 30), (30, 40), (40, 50)], [(10, 16), (20, 26), (30, 36), (40, 46)]],
```